editor/src/core/model/element-template-utils.spec.tsx

```diff
@@ -739,3 +749,245 @@ describe('rearrangeJsxChildren', () => {
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/mapped-root/mapped-child',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/ternary-true-root',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/ternary-true-root/ternary-true-child',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/ternary-false-root',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/ternary-false-root/ternary-false-child',
+    ])
+  })
+
+  it('conditional expressions', () => {
+    const projectFile = getParseSuccessForStoryboardCode(
+      makeTestProjectCodeWithSnippet(`
+        <div style={{ ...props.style }} data-uid='aaa'>
+          <div data-uid='parent' >
+            <div data-uid='child-d' />
+            {true ? 
+              (
+                <div data-uid='ternary-true-root'>
+                  <div data-uid='ternary-true-child' />
+                </div> 
+              ) : (
+                <div data-uid='ternary-false-root'>
+                  <div data-uid='ternary-false-child' />
+                </div> 
+              )
+            }
+            <div data-uid='child-b' />
+            <div data-uid='child-a' />
+          </div>
+        </div>
+      `),
+    )
+
+    expectElementAtPathHasMatchingUIDForPaths(projectFile, [
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/ca0',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/ca0/ternary-true-root',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/ca0/ternary-true-root/ternary-true-child',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/ca0/ternary-false-root',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/ca0/ternary-false-root/ternary-false-child',
+    ])
+  })
+
+  it('conditional expressions with children that are JSXAttribute', () => {
+    const projectFile = getParseSuccessForStoryboardCode(
+      makeTestProjectCodeWithSnippet(`
+        <div style={{ ...props.style }} data-uid='aaa'>
+          <div data-uid='parent' >
+            <div data-uid='child-d' />
+            {true ? 
+              (
+                "hello"
+              ) : (
+                "world"
+              )
+            }
+            <div data-uid='child-b' />
+            <div data-uid='child-a' />
+          </div>
+        </div>
+      `),
+    )
+
+    expectElementAtPathHasMatchingUIDForPaths(projectFile, [
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent',
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/ca0',
+    ])
+
+    // !!! what to do with the then-case and else-case "uid" behaviors? how do we proceed from here?
+
+    expectElementFoundNull(projectFile, [
+      'utopia-storyboard-uid/scene-aaa/app-entity:aaa/parent/ca0/then-case',
```

Comment:
⚠️ When asking for `aaa/parent/ca0/then-case`, findJSXElementChildAtPath will return the conditional itself at `aaa/parent/ca0`!!

I believe this is not the way it should be. I think if we target either branch of the conditional _and_ the branch doesn't have a JSXElementChild to return, we should return null. 

@ruggi do you have any thoughts about this? 

Zooming out, the then-case and else-case code has been growing in the last week with new navigator features and lookup functions. Is the behavior internally consistent? What would happen if findJSXElementChildAtPath starts returning (I believe correctly) null for the cases when the element at the then-case or else-case is, in fact, NOT a jsxElementChild?

Reply:
> When asking for aaa/parent/ca0/then-case, findJSXElementChildAtPath will return the conditional itself at aaa/parent/ca0
this should definitely not be the case :)